tests/ragger/app/client.py

```diff
@@ -156,3 +194,57 @@ def eip712_filtering_show_field(self, name: str, sig: bytes):
         with self._send(self._cmd_builder.eip712_filtering_show_field(name, sig)):
             pass
         assert self._recv().status == 0x9000
+
+    def send_fund(self,
+                  bip32_path: list[Optional[int]],
+                  nonce: int,
+                  gas_price: int,
+                  gas_limit: int,
+                  to: bytes,
+                  amount: float,
+                  chain_id: int,
+                  screenshot_collection: str = None):
+        for chunk in self._cmd_builder.send_fund(bip32_path,
+                                                 nonce,
+                                                 gas_price,
+                                                 gas_limit,
+                                                 to,
+                                                 amount,
+                                                 chain_id):
+            with self._send(chunk):
+                nav_ins = NavIns(NavInsID.RIGHT_CLICK)
+                final_ins = [ NavIns(NavInsID.BOTH_CLICK) ]
+                target_text = "and send"
+                if screenshot_collection:
+                    self._nav.navigate_until_text_and_compare(nav_ins,
+                                                              final_ins,
+                                                              target_text,
+                                                              ROOT_SCREENSHOT_PATH,
+                                                              screenshot_collection)
+                else:
+                    self._nav.navigate_until_text(nav_ins,
+                                                  final_ins,
+                                                  target_text)
+            assert self._recv().status == 0x9000
```

Comment:
I would prefer this function to yield when the final chunk is sent, and the final navigation is done in the test code. This would give more granularity if for example we want to test the refusal
That's also more consistent with how we designed tests until now

If you think it's a good idea but you don't have time to do it I volunteer.

Also don't do the `self._recv().status == 0x9000`, ragger would raise anyway if we receive an error code, except if the test configures the backend to not raise in which case we actually don't want to raise

Reply:
The choice to have the navigation done in the client is explained in other comments.

I removed the status word check. :+1: